proposal/color-4-new-spaces.md

```diff
@@ -0,0 +1,1984 @@
+
+    > This basic restriction can be applied to all spaces. Further channel
+    > restrictions are enforced by the normalization step for known spaces.
+
+  * If `key` is a string in the format `channel<integer>`, set `key` to the
+    value of `<integer>`.
+
+    > This allows e.g. `color.change($color, $channel1: 0.25)` for changing
+    > color channels in unknown color spaces.
+
+  * If `key` is not the name or index of a channel in `channels`, throw an
+    error.
+
+  * Set the corresponding `channel`'s value in `channels` to `new`.
+
+* If `space` is a [known color space], set `channels` to the result of
+  [normalizing] `channels` in `space`.
+
+* Let `new` be a color in color space `space`, with `channels` channels, and an
+  alpha of `alpha`.
+
+* Return the result of [converting] `new` into `origin-space`.
+
+### `color.adjust()`
+
+```
+adjust($color, $args...)
+```
+
+This function is also available as a global function named `adjust-color()`.
+
+* If `$color` is not a color, throw an error.
+
+* If any item in `$args` is not a keyword argument, throw an error.
+
+* Let `origin-space` be `$color`'s color space.
+
+* If the keyword argument `$space` is specified in `$args`:
+
+    * Let `space` be the result of [looking up a known color space] named
+      `$space`.
+
+    * If `space` is not a [known color space], and `space != origin-space`,
+      throw an error.
+
+      > We can make adjustments in an unknown space, but we can't do conversion.
+
+    * Let `color` be the result of [converting] `$color` to `space`.
+
+* Otherwise:
+
+  * Let `space` be `origin-space`.
+
+  * Let `color` be the value of `$color`.
+
+* Let `alpha` be `color`'s alpha property.
+
+* If the keyword argument `$alpha` is specified in `$args`:
+
+  * If `$alpha` is not a number, throw an error.
+
+  * Let `new-alpha` be the result of [percent-converting] `$alpha` with a max
+    of 1.
+
+  * Set `alpha` to the value of `new-alpha + alpha` clamped between 0 and 1.
+
+* Let `channel-args` be the remaining keyword arguments in `$args`, not
+  including `$space` or `$alpha` arguments.
+
+* Let `channels` be a list of the `color`'s channels.
+
+* For each keyword `key` and value `adjust` in `channel-args`:
+
+  * If `key` is a string in the format `channel<integer>`, set `key` to the
+    value of `<integer>`.
+
+    > This allows e.g. `color.adjust($color, $channel1: 0.25)` for adjusting
+    > color channels in unknown color spaces.
+
+  * If `key` is not the name or index of a channel in `channels`, throw an
+    error.
+
+  * Set `adjust` to the result of [validating] `adjust` as `key` channel in
+    `space`.
+
+    > This ensures that `adjust` is percent-converted when necessary, and
+    > throws errors when the adjustment is not a valid value for the space.
+
+  * Set the `channel` with name or index of `key` in `channels` to
+    `channel + adjust`, treating values of `none` as `0`.
```

Comment:
Since `none` indicates that the channel value is _irrelevant_, should we just preserve it as-is even if an adjustment is made to it?

Reply:
I would expect `adjust()` and `scale()` to have the same behavior here. Likely choosing between:

- Maintain `none` as an irrelevant channel. This strikes me as potentially confusing for authors, since it results in no change and no notification?
- Adjust/scale from `0`.
- Throw an error.

CSS color 5 has a similar situation, where color channels can be adjusted in a relative color syntax. I don't see it clearly defined in that spec what should happen when missing channels are adjusted. I think it would be great to follow their lead if we're able? I've reached out to Chris, and can also open an issue in CSS. My expectation would be either adjusting-from-0, or considering it an invalid operation.